src/main/java/core/basesyntax/service/DataConverter.java

```
@@ -0,0 +1,8 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public interface DataConverter {
+    List<FruitTransaction> convertToTransaction(List<String> rawLines);
```

The checklist requires handling null parameters and failing fast with informative exceptions. Currently DataConverterImpl returns an empty list for `rawLines == null`, which violates that expectation. Ensure implementations throw FruitShopException for null `rawLines` and include offending line in parse error messages. See DataConverterImpl for current implementation to update.